semgrep/semgrep/output.py

```diff
@@ -307,9 +307,9 @@ def close(self) -> None:
         elif self.semgrep_structured_errors:
             # make a simplifying assumption that # errors = # files failed
             # it's a quite a bit of work to simplify further because errors may or may not have path, span, etc.
-            error_stats = (
-                f"{len(self.semgrep_structured_errors)} files could not be analyzed"
-            )
+            num_errors = len(self.semgrep_structured_errors)
+            plural = "s" if num_errors > 1 else ""
+            error_stats = f"found problems analyzing {num_errors} file{plural}"
             final_error = self.semgrep_structured_errors[-1]
```

Comment:
I think with this hacky assumption to have final_error be the last element of semgrep_structured_errors and CoreWarning having a 0 return code and https://github.com/returntocorp/semgrep/blob/develop/semgrep/semgrep/__main__.py#L25 I think there might be a scenario when semgrep should be exiting with non-zero exit code but instead has a 0 exit code?

Reply:
Good catch, I don't have an example of that but it seems possible to me. I'll give it a try at fixing it.